scripts/build/gen_isr_tables.py

```diff
@@ -131,12 +131,65 @@ def parse_args():
 #endif
 """
 
+sh_isr_function = "void __attribute__((section(\".shared_isr\")))shared_isr_"
+sh_isr_args = "const size_t shared_isr_"
+sh_isr_args_section = " __attribute__((section(\".shared_isr_args\")))"
+
 def get_symbol_from_addr(syms, addr):
     for key, value in syms.items():
         if addr == value:
             return key
     return None
 
+def write_shared_direct_isr_prototypes(fp, nv, shi):
+    for i in range(nv):
+        if len(shi[i]) > 0:
+            fp.write(sh_isr_function + "direct_" + str(i) + "(void);\n")
```

Comment:
Are you sure this doesn't need a `__attribute__ ((interrupt))`. Did you verify in assembly what's generated by the compiler?

Reply:
I tested the direct ISR and it worked but I agree with you about adding __attribute__ ((interrupt))